### PR TITLE
refactor(neon_news, neon_notes, neon_notifications): replace long press actions with swipe-to-dismiss 

### PR DESCRIPTION
**Pull request by satvik2131**
#2158 
## Implemented swipe-to-dismiss feature in : 
1. Neon News 
   - Articles (swipe to mark read/unread)
   - Folders
   - Feeds

2. Neon Notifications 
3. Neon notes

### CODE REVIEW

File: packages/neon/neon_news/lib/src/widgets/folders_view.dart [hunk 1]
```diff
@@ -60,85 +61,91 @@ class NewsFoldersView extends StatelessWidget {
...
+                      value: NewsFolderAction.delete,
+                      child: Text(NewsLocalizations.of(context).actionDelete),
+                    ),
+                    PopupMenuItem(
+                      value: NewsFolderAction.rename,
+                      child: Text(NewsLocalizations.of(context).actionRename),
+                    ),
+                  ],
+                  onSelected: (action) async {
+                    if (action case NewsFolderAction.rename) {
```
Comment (provokateurin) [inline packages/neon/neon_news/lib/src/widgets/folders_view.dart:117]: Revert

**Review by provokateurin — CHANGES_REQUESTED** [review]

**Comment by provokateurin** [comment]
Thanks for the work btw :)

**Comment by satvik2131** [comment]
> Thanks for the work btw :)

most welcome , will be updating the changes soon, and btw you are literally very active , thanks for the feedback and guidance :)

File: packages/neon/neon_news/lib/src/widgets/feeds_view.dart [hunk 2]
```diff
@@ -56,122 +56,129 @@ class NewsFeedsView extends StatelessWidget {
...
+                PopupMenuButton(
+                  itemBuilder: (context) => [
+                    PopupMenuItem(
+                      value: NewsFeedAction.showURL,
+                      child: Text(NewsLocalizations.of(context).feedShowURL),
+                    ),
+                    // PopupMenuItem(
+                    //   value: NewsFeedAction.delete,
+                    //   child: Text(NewsLocalizations.of(context).actionDelete),
+                    // ),
```
[the remark this answers is not in the document]
Comment (satvik2131) [reply packages/neon/neon_news/lib/src/widgets/feeds_view.dart:113]: should I revert the changes , sorry , I didn't get it properly ?

[the remark this answers is not in the document]
Comment (provokateurin) [reply packages/neon/neon_news/lib/src/widgets/feeds_view.dart:113]: Ah the context is too wide. Only the commented out code should be reverted

**Review by provokateurin — APPROVED** [review]
Thank you so much!
You need to fix the DCO and please name the commit something like `refactor(neon_news,neon_notes,neon_notifications): Replace long press actions with swipe-to-dismiss`

**Comment by provokateurin** [comment]
You only changed the PR title, not the commit message. Also please don't mention the issue, you can just put it in the PR description to refer to the issue.

**Comment by provokateurin** [comment]
Please checkout the issues the CI found. You can run `melos run format` to fix the formatting problems, but the analyzer issues need to be fixed as well.

File: packages/neon/neon_notifications/lib/src/widgets/notification.dart [hunk 3]
```diff
@@ -22,45 +22,53 @@ class NotificationsNotification extends StatelessWidget {
...
+              .toList(),
+        ),
+        leading: NeonUriImage(
+          account: NeonProvider.of<Account>(context),
+          uri: Uri.parse(notification.icon!),
+          size: const Size.square(largeIconSize),
+          svgColorFilter: ColorFilter.mode(Theme.of(context).colorScheme.primary, BlendMode.srcIn),
+        ),
+        onTap: notification.link.isNotEmpty ? () => context.go(notification.link) : null,
+        onLongPress: onDelete,
```
Comment (provokateurin) [inline packages/neon/neon_notifications/lib/src/widgets/notification.dart:70]: So now we have two ways to delete the notification? That's a bit strange UX wise and

File: packages/neon/neon_notes/lib/src/widgets/notes_view.dart [hunk 4]
```diff
@@ -56,68 +56,78 @@ class NotesView extends StatelessWidget {
...
+        direction: DismissDirection.startToEnd,
+        background: Container(color: Colors.red),
+        confirmDismiss: (direction) async {
           final result = await showConfirmationDialog(
             context: context,
             title: NotesLocalizations.of(context).noteDeleteConfirm(note.title),
           );
+
           if (result) {
             bloc.deleteNote(note.id);
```
Comment (provokateurin) [inline packages/neon/neon_notes/lib/src/widgets/notes_view.dart:70]: I Just looked closer at the docs of Dismissible and the action itself should be done in onDismissed and not in confirmDismiss.

File: packages/neon/neon_news/lib/src/widgets/folders_view.dart [hunk 5]
```diff
@@ -60,84 +60,97 @@ class NewsFoldersView extends StatelessWidget {
...
-            Center(
-              child: Text(
-                feedCount.toString(),
-                style: TextStyle(
-                  color: Theme.of(context).colorScheme.onPrimary,
+
+    return Dismissible(
+      key: Key(folder?.id.toString() ?? '0'),
+      direction: DismissDirection.startToEnd,
+      confirmDismiss: (direction) async {
```
Comment (provokateurin) [inline packages/neon/neon_news/lib/src/widgets/folders_view.dart:67]: onDismissed

File: packages/neon/neon_news/lib/src/widgets/feeds_view.dart [hunk 6]
```diff
@@ -56,121 +56,129 @@ class NewsFeedsView extends StatelessWidget {
...
-            if (feed.updateErrorCount > 0)
-              IconButton(
-                onPressed: () async => showAdaptiveDialog<void>(
-                  context: context,
-                  builder: (context) => NewsFeedUpdateErrorDialog(
-                    feed: feed,
+      Dismissible(
+        key: Key(feed.id.toString()),
+        direction: DismissDirection.startToEnd,
+        confirmDismiss: (direction) async {
```
Comment (provokateurin) [inline packages/neon/neon_news/lib/src/widgets/feeds_view.dart:62]: onDismissed

File: packages/neon/neon_news/lib/src/widgets/articles_view.dart [hunk 7]
```diff
@@ -139,145 +139,152 @@ class _NewsArticlesViewState extends State<NewsArticlesView> {
...
-          children: [
-            Container(
-              margin: const EdgeInsets.only(
-                top: 8,
-                bottom: 8,
-                right: 8,
+      Dismissible(
+        key: Key(article.id.toString()),
+        direction: DismissDirection.startToEnd,
+        confirmDismiss: (direction) async {
```
Comment (provokateurin) [inline packages/neon/neon_news/lib/src/widgets/articles_view.dart:145]: onDismissed

**Review by provokateurin — CHANGES_REQUESTED** [review]

**Review by provokateurin — APPROVED** [review]

**Pull request merged**